File: week-2/class-problems/http-server-in-different-language/main.go
package main

import (
	"fmt"
	"net/http"
	"github.com/gorilla/mux"
)

func handler(w http.ResponseWriter, r *http.Request) {
	fmt.Fprint(w, "Hello from Go with Gorilla Mux!\n")
}

func main() {
	r := mux.NewRouter()
	r.HandleFunc("/", handler)

	PORT := 3000
	fmt.Printf("Server running at http://localhost:%d/\n", PORT)
	http.Handle("/", r)
	http.ListenAndServe(fmt.Sprintf(":%d", PORT), nil)
}

File: week-2/class-problems/http-server-in-different-language/Main.java
import java.io.IOException;
import java.io.OutputStream;
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpServer;

public class Main {
    public static void main(String[] args) throws IOException {
        HttpServer server = HttpServer.create(new java.net.InetSocketAddress(3000), 0);
        server.createContext("/", new MyHandler());
        server.setExecutor(null);
        server.start();
        System.out.println("Server running at http://localhost:3000/");
    }

    static class MyHandler implements HttpHandler {
        @Override
        public void handle(HttpExchange t) throws IOException {
            String response = "Hello from Java!\n";
            t.sendResponseHeaders(200, response.length());
            OutputStream os = t.getResponseBody();
            os.write(response.getBytes());
            os.close();
        }
    }
}

// java Main
